File: mddProject/src/app/feature/articles/interfaces/article.ts
export interface Article {
    id: number;
    title: string ;
    content: string ;
    createdAt: Date ;
    updatedAt: Date ;
    userId: number ;
    themeId: number | null ;
    authorName: string ;
    themeTitle: string ;
    comments: Comment[]; // Commentaires de l'article;  // Liste des commentaires associés à l'article
}

export interface Comment {


    id: number;
    content: string;
    userId: number;
    authorName: string;
    articleId: number;

}

File: back/src/main/java/com/mdd/pocmdd/models/Article.java
package com.mdd.pocmdd.models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.List;
import java.time.LocalDateTime;

@Data
@Entity
@NoArgsConstructor
@Table(name = "articles")
public class Article {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 255)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Column(name = "created_at", nullable = false, unique = true)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", unique = true)
    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "theme_id", nullable = false)
    private Theme theme;

    @OneToMany(mappedBy = "article", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Comment> comments; // Liste des commentaires associés à l'article
}
